### DISCUSSION TITLE
Can we get joint axis?

### DISCUSSION THREAD

**Discussion by keiohta** [haosulab/SAPIEN#71] [Q&A] [answered]
Hi author, I want to know a joint axis but haven't figured out how to do that. It seems [provided methods](https://github.com/haosulab/SAPIEN/blob/master/python/pysapien_content.hpp#L990-L1074) do not include such functionality, so opening this thread. So, is there any possible way to get a joint axis from a `PyJoint` object?

I imagine if a URDF is like:
```
    <joint name="joint_1" type="revolute">
        <origin xyz="0 0 0" />
        <axis xyz="0 0 1" />
        <child link="link_1" />
        <parent link="link_0" />
        <limit lower="0.0" upper="3.141592653589793" />
    </joint>
```
then I can get `(0, 0, 1)` when we call a method like `joint.get_axis()`.

Thanks in advance for your continuous support.

**Comment by fbxiang** [comment, marked as answer]
You can use `joint.get_parent_link().pose * joint.get_pose_in_parent()` to get the frame of this joint. The x axis of this frame is the joint direction, and the position of the frame is the joint origin. For example, you can extract the direction by `pose.to_transformation_matrix()[:3,:3] @ [1,0,0]`

**Reply by keiohta** [reply]
Your answer completely solved my issue. Thank you!!
